src/backend/executor/execMain.c

```diff
@@ -1998,12 +1998,17 @@ InitPlan(QueryDesc *queryDesc, int eflags)
 		{
 			/*
 			 * A subplan will never need to do BACKWARD scan nor MARK/RESTORE.
-			 *
-			 * GPDB: We always set the REWIND flag, to delay eagerfree.
 			 */
 			sp_eflags = eflags
 				& (EXEC_FLAG_EXPLAIN_ONLY | EXEC_FLAG_WITH_NO_DATA);
-			sp_eflags |= EXEC_FLAG_REWIND;
+
+			/*
+			 * GPDB: For ORCA, we always set the REWIND flag to delay
+			 * eagerfree. However, planner can generate InitPlans, which are
+			 * non-rewindable.
+			 */
+			if (bms_is_member(subplan_id, plannedstmt->rewindPlanIDs))
```

Comment:
In the original patch `subplan_idx + 1` is used. Why is it changed? Shouldn't it be reflected in the commit description (in the section with changes comparing to the original commit)?

Reply:
In GPDB7, the `subplan_id` variable is assigned to 1 on initialization, and in some places it is used as `subplan_id - 1`. In GPDB6, it is assigned to 0 before the usage and is used as is in places I pointed out before.